AV 1008 - Simulation configuration files

mpkarpov-ui opened the pull request:
This ticket will change the primary way that simulations define simulation-specific data from a `properties.py` file to a `.yaml` config. The README for the changes is included on the branch, so please reach out if it isn't fully comprehensible.

Program on branch should function identically to `main`.

Dependencies:
- PyYaml -- `pip install pyyaml`

rbhog commented:
ik u didn't request a review from me but just wanted to say this is sick, bless ur soul. loading in a new object that represents the properties across each class isn't the most performant in terms of memory usage, is there any way to optimize it so we're loading from the yamls as little as possible?

mpkarpov-ui commented:
I've added you as a reviewer (the more the merrier!) -- As for loading in a new object, I think this might have just been a documentation issue, because it used to be that you had to call `data_loader.load_config(filepath)` in every class that you wanted to have a config, but in one of the more recent commits I changed it so that `data_loader` instead does a config load the first time it is imported, and exposes a `config` variable so it can be accessed by `dataloader.config`. Now correct me if I'm wrong (im not a python expert), but because of Python's "load-once" rule, `data_loader` shouldn't be run more than once, and thus the config load should only happen once, right?

I made the change so you can share the same config between multiple scripts and also for optimization purposes. `data_loader` *does* still expose the `load_config()` method but I made it (for now) print an error if the same process loads more than one config.

ps: Upon looking into the documentation I noticed I haven't written some of this stuff in so I'll make another commit

mpkarpov-ui commented:
I've made the doc change, please lmk of any other issues

EvanY14 reviewed (commented):
It looks really good to me. I haven't had a chance to pull it and run it locally yet, but based on the Monte Carlo results it looks like the changes work well. Great work!

EvanY14 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [[AV-1008] Simulation config files](https://trello.com/c/YfJjViBU/1008-av-1008-simulation-config-files)

EvanY14 commented on Simulation/6DOF_RK4/properties/templates/motors.yaml:
```diff
@@ -0,0 +1,23 @@
+  <<: *simulated_motor
+  impulse: 20145.7
+  motor_mass: 14.826
+  delay: 0
```
Change to 60

mpkarpov-ui replied:
Done

EvanY14 commented on Simulation/6DOF_RK4/properties/templates/motors.yaml:
```diff
@@ -0,0 +1,23 @@
+  <<: *simulated_motor
+  impulse: 17907.0
+  motor_mass: 10.7
+  delay: 0
```
Change to 60

mpkarpov-ui replied:
Done
